url20b0f880a5_jgraph_legacy_jgraph5_tgz-pJ8-org_jgraph_example_HelloWorldJ8/src/com/mxgraph/util/mxCurve.java

```diff
@@ -0,0 +1,1244 @@
+		double candidateOffX2 = (absPoint.getX() + segNormY * orthogonalOffset)
+				- endSegPt.getX();
+		double candidateOffY2 = (absPoint.getY() - segNormX * orthogonalOffset)
+				- endSegPt.getY();
+
+		double candidateDist1 = (candidateOffX1 * candidateOffX1)
+				+ (candidateOffY1 * candidateOffY1);
+		double candidateDist2 = (candidateOffX2 * candidateOffX2)
+				+ (candidateOffY2 * candidateOffY2);
+
+		double orthOffsetPointX = 0;
+		double orthOffsetPointY = 0;
+
+		if (candidateDist2 < candidateDist1)
+		{
+			orthogonalOffset = -orthogonalOffset;
+		}
+
+		orthOffsetPointX = absPoint.getX() - segNormY * orthogonalOffset;
+		orthOffsetPointY = absPoint.getY() + segNormX * orthogonalOffset;
+
+		double distAlongEdge = 0;
+		double cartOffsetX = 0;
+		double cartOffsetY = 0;
+
+		// Don't compare for exact equality, there are often rounding errors
+		if (Math.abs(closestSegDistSq - lineDistSq) > 0.0001)
+		{
+			// The orthogonal offset does not move the point onto the
+			// segment. Work out an additional cartesian offset that moves
+			// the offset point onto the closest end point of the
+			// segment
+
+			// Not exact distances, but the equation holds
+			double distToStartPoint = Math.abs(orthOffsetPointX
+					- startSegPt.getX())
+					+ Math.abs(orthOffsetPointY - startSegPt.getY());
+			double distToEndPoint = Math
+					.abs(orthOffsetPointX - endSegPt.getX())
+					+ Math.abs(orthOffsetPointY - endSegPt.getY());
+			if (distToStartPoint < distToEndPoint)
+			{
+				distAlongEdge = currentIntervals[closestSegment];
+				cartOffsetX = orthOffsetPointX - startSegPt.getX();
+				cartOffsetY = orthOffsetPointY - startSegPt.getY();
+			}
+			else
+			{
+				distAlongEdge = currentIntervals[closestSegment + 1];
+				cartOffsetX = orthOffsetPointX - endSegPt.getX();
+				cartOffsetY = orthOffsetPointY - endSegPt.getY();
+			}
+		}
+		else
+		{
+			// The point, when orthogonally offset, lies on the segment
+			// work out what proportion along the segment, and therefore
+			// the entire curve, the offset point lies.
+			double segmentLen = Math.sqrt((endSegPt.getX() - startSegPt.getX())
+					* (endSegPt.getX() - startSegPt.getX())
+					+ (endSegPt.getY() - startSegPt.getY())
+					* (endSegPt.getY() - startSegPt.getY()));
+			double offsetLen = Math.sqrt((orthOffsetPointX - startSegPt.getX())
+					* (orthOffsetPointX - startSegPt.getX())
+					+ (orthOffsetPointY - startSegPt.getY())
+					* (orthOffsetPointY - startSegPt.getY()));
+			double proportionAlongSeg = offsetLen / segmentLen;
+			double segProportingDiff = currentIntervals[closestSegment + 1]
+					- currentIntervals[closestSegment];
+			distAlongEdge = currentIntervals[closestSegment]
+					+ segProportingDiff * proportionAlongSeg;
+		}
+
+		if (distAlongEdge > 1.0)
+		{
+			distAlongEdge = 1.0;
+		}
+
+		return new mxRectangle(distAlongEdge, orthogonalOffset, cartOffsetX,
+				cartOffsetY);
+	}
+
+	/**
+	 * Creates the actual 
+	 */
+	protected void createCoreCurve()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 26. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 876-898 into a separate method.